### ISSUE TITLE
Fix CSS issue in mxnet docs site header

### ISSUE THREAD

**Issue by sandeep-krishnamurthy** [apache/incubator-mxnet#3531]
**Issue**
- When browser window size is reduced, header gets distorted.
- After reducing size to very small size, headers will not even be visible.

**Reproduce the issue**
- open http://mxnet.io
- Keep shrinking (reducing) size of web page in your browser
- Observe that first headers get distorted later even disappears.

**Possible solutions**
**Solution 1:**  Fix minimum width so all headers will be visible and not get distorted.
**Solution 2:**  Change CSS for headers such that it gets elongated and all headers are listed one below other when users shrinks web page size.

**Comment by sandeep-krishnamurthy** [comment]
For example, open https://www.tensorflow.org/versions/r0.11/how_tos/index.html and start reducing the web page size. It reduces till acceptable width and donot change after that.

Headers will all get re-arranged.

This is achieved by:
1. Making header position relative
2. Fixing header <div> min width (Post that scroll bars comes in to picture)

**Comment by pluskid** [comment]
On small vertical screens (e.g. a phone), the common behavior that other websites adopt is that the menu is collapsed into a button that the user could click to expand and navigate. Currently it is impossible to navigate our doc on a phone.

**Comment by pluskid** [comment]
Also the style of the doc needs some fixing. For example, currently the `parameters` and `returns` are not very clearly separated. Take the doc for [infer_shape](http://mxnet.io/api/python/symbol.html#mxnet.symbol.Symbol.infer_shape) for example, it is very difficult to tell which one is the return value and which one is parameters. I'm not sure if we have any other small glitches here and there. If we do not have front-end developer, a good solution might be to clone one of the widely used theme and then only change the color scheme to match our website. Some good documentation include [numpy](http://docs.scipy.org/doc/numpy/reference/generated/numpy.vdot.html#numpy.vdot), [keras](https://keras.io/layers/core/#spatialdropout2d), [tensorflow](https://www.tensorflow.org/versions/r0.11/api_docs/python/math_ops.html#sqrt), etc.

**Comment by pluskid** [comment]
Also current the indentation in the example code of the doc string are somehow removed. See [here](http://mxnet.io/api/python/symbol.html#mxnet.symbol.Flatten) for example.

**Comment by sandeep-krishnamurthy** [comment]
1. Indentation of headers, bullet points etc. have issues. see this page for example - http://mxnet.io/get_started/setup.html

2. We need better font size and style

3. Heading after 5th nesting i.e., "#####" will have very small font size. Actually smaller than other content font size. Need to fix this.

**Comment by kevinthesun** [comment]
I took a look at tensor flow website. The design for navigation bar is not so nice especially for smart phone screens. I agree that maybe we should consider to collapse the nav bar into a dropdown menu or a hamburger button.

**Comment by pluskid** [comment]
Another mysterious issue is that the "Regression Test" header is eaten in the docstring. For example, if you look at [python/mxnet/symbol_doc.py](https://github.com/dmlc/mxnet/blob/master/python/mxnet/symbol_doc.py#L81), to check the doc for `Dropout`. Comparing with what doc we currently have [online](http://mxnet.io/api/python/symbol.html#mxnet.symbol.Dropout). The "Regression Test" header somehow disappeared.

**Comment by kevinthesun** [comment]
I have updated nav bar and need to push my branch. can someone add me to this project?

**Comment by kevinthesun** [comment]
Also for the sidebar of TOC issue, it is due to that margin left of container is set to auto. So what is the effect we need for TOC sidebar? If we just need a fixed-width sidebar, we just need to change the margin left value.

**Comment by pluskid** [comment]
@kevinthesun Thanks a lot! We do not directly push to the main repo. You can make a clone and then submit a pull request. See [the mxnet doc](http://mxnet.io/community/contribute.html#submit-pull-request) and [the github doc](https://help.github.com/articles/about-pull-requests/).

**Comment by kevinthesun** [comment]
Another issue is that "Setup and Installation" is linked to the absolute address "http://mxnet.io/get_started/setup.html". Should it be linked to relative address "../get-started/setup/html"?

**Comment by pluskid** [comment]
Yes, relative path is always preferred.

**Comment by kevinthesun** [comment]
How to modify these dynamic contents?

**Comment by pluskid** [comment]
Docs are generated from [the markdown files in docs directory](https://github.com/dmlc/mxnet/tree/master/docs).

**Comment by kevinthesun** [comment]
Currently I'm working on my local branch. I need to push it to remote and make a new pull request right? But it requires github username and password.

**Comment by pluskid** [comment]
@kevinthesun You will need to make a clone of your own repo, add a `remote` to your own clone, push to your own clone and submit a pull request from there.

**Comment by pluskid** [comment]
I found that the [devdocs](http://devdocs.io/) website has a very nice style for documentations. It also works well with mobile devices. If someone would like to adapt that to MXNet documentation, the source code of devdocs is [hosted here](https://github.com/Thibaut/devdocs).

**Comment by kevinthesun** [comment]
I have opened a new pull request. If you have chance to take a look at the changes and test them, it would be nice!

**Comment by kevinthesun** [comment]
Do we need to immigrate the whole site to this template? Or we just modify the current version?

**Comment by pluskid** [comment]
@kevinthesun Thanks! That is one of the reference style that (I think) looks good and also works one mobile phone. My guess is starting from an existing working stylesheet and adapt that to our existing theme (e.g. overall color scheme and make the front page works properly) is easier than modify our existing style trying to mimic others, because it can be hard to cover all aspects and glitches and inconsistency throughout browsers and devices. But I'm not an expert in front end so my words might not make sense. :)

**Comment by kevinthesun** [comment]
@pluskid  It is a good suggestion we should use a template, but right now since most contents are completed, I prefer modifying the existing website. I have fixed the index page to make it work well for mobile device. The change has been merged into master so you can check it out.

**Comment by kevinthesun** [comment]
Can we close this?

**Comment by szha** [comment]
Issues at the top seem to have been fixed.
